Competition2017/src/competition/subsystems/shooter_belt/commands/RunBeltIfWheelAtSpeedCommand.java

```diff
@@ -4,20 +4,18 @@
 
 import competition.subsystems.shooter_belt.ShooterBeltSubsystem;
 import competition.subsystems.shooter_wheel.ShooterWheelSubsystem;
-import xbot.common.command.BaseCommand;
-
-public class RunBeltIfWheelAtSpeedCommand extends BaseCommand {
+    
+public class RunBeltIfWheelAtSpeedCommand extends BaseShooterBeltCommand {
 
     private static Logger log = Logger.getLogger(RunBeltIfWheelAtSpeedCommand.class);
     
-    final ShooterBeltSubsystem beltSubsystem;
-    public ShooterWheelSubsystem shooterWheelSubsystem;
-    final double targetBeltSpeed;
-    
-    public RunBeltIfWheelAtSpeedCommand(ShooterBeltSubsystem beltSubsystem){
-        this.beltSubsystem = beltSubsystem;
-        this.requires(beltSubsystem);
-        targetBeltSpeed = beltSubsystem.getTargetSpeed();
+    protected final ShooterWheelSubsystem shooterWheelSubsystem;
+    private double targetBeltSpeed;
+
+    public RunBeltIfWheelAtSpeedCommand(ShooterBeltSubsystem shooterBeltSubsystem, ShooterWheelSubsystem shooterWheelSubsystem) {
+        super(shooterBeltSubsystem);
+        this.shooterWheelSubsystem = shooterWheelSubsystem;
+        targetBeltSpeed = shooterBeltSubsystem.getTargetSpeed();
```

Comment:
This just gets the most recent set target speed, which could be anything (e.g. 0). This needs to read directly from some new DoubleProperty on the BeltSubsystem like `BeltIntakeSpeed`